src/main/java/io/reactivex/internal/operators/flowable/FlowableCache.java

```diff
@@ -50,16 +50,24 @@ public FlowableCache(Flowable<T> source, int capacityHint) {
     protected void subscribeActual(Subscriber<? super T> t) {
         // we can connect first because we replay everything anyway
         ReplaySubscription<T> rp = new ReplaySubscription<T>(t, state);
-        state.addChild(rp);
-
         t.onSubscribe(rp);
 
+        boolean doReplay = true;
```

Comment:
The old logic could have made the `Subscriber` visible too early and thus it could receive an `onComplete` before an `onSubscribe`.